File: output/GPT4o-Mini/xCodeEval_4000/C++/C/769430fb6a97a234c854f4c874271503.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

#define ll long long
#define MOD 1000000007

ll dp[20][2][2][10][10];
ll memo[20][2][2][10][10];
ll n;
ll num[20];

ll solve(ll ind, ll ti, ll st, ll f, ll pd) {
    if (ind == n) {
        return (st > 0 && f == pd);
    }
    if (memo[ind][ti][st][f][pd]) return dp[ind][ti][st][f][pd];
    
    ll res = 0;
    memo[ind][ti][st][f][pd] = 1;
    
    if (ti) {
        for (ll i = 0; i < num[ind]; i++) {
            if (i == 0) res += solve(ind + 1, 0, st, f, i);
            if (i > 0) {
                if (st == 0) res += solve(ind + 1, 0, 1, i, i);
                else res += solve(ind + 1, 0, st, f, i);
            }
        }
        if (num[ind] == 0) res += solve(ind + 1, 1, st, f, 0);
        else {
            if (st == 0) res += solve(ind + 1, 1, 1, num[ind], num[ind]);
            else res += solve(ind + 1, 1, st, f, num[ind]);
        }
    } else {
        for (ll i = 0; i < 10; i++) {
            if (i == 0) res += solve(ind + 1, 0, st, f, i);
            if (i > 0) {
                if (st == 0) res += solve(ind + 1, 0, 1, i, i);
                else res += solve(ind + 1, 0, st, f, i);
            }
        }
    }
    return dp[ind][ti][st][f][pd] = res;
}

int main() {
    ll l, r, k;
    scanf("%lld %lld", &l, &r);
    
    ll idx = 0;
    while (r) {
        num[idx++] = r % 10;
        r /= 10;
    }
    n = idx;
    ll ans = 0;
    memset(dp, 0, sizeof(dp));
    memset(memo, 0, sizeof(memo));
    ans = solve(0, 1, 0, 0, 0);
    
    idx = 0;
    memset(dp, 0, sizeof(dp));
    memset(memo, 0, sizeof(memo));
    k = l;
    while (l) {
        num[idx++] = l % 10;
        l /= 10;
    }
    n = idx;
    ans -= solve(0, 1, 0, 0, 0);
    
    idx = 0;
    while (k) {
        num[idx++] = k % 10;
        k /= 10;
    }
    ans += (num[0] == num[idx - 1]);
    printf("%lld\n", ans);
    
    return 0;
}

File: output/GPT4o-Mini/xCodeEval_4000/C++/Go/769430fb6a97a234c854f4c874271503.go
package main

import (
	"fmt"
	"math"
)

const mod = 1e9 + 7

var dp [20][2][2][10][10]int64
var memo [20][2][2][10][10]int64
var num []int64
var n int64

func solve(ind int64, ti int64, st int64, f int64, pd int64) int64 {
	if ind == n {
		if st > 0 && f == pd {
			return 1
		}
		return 0
	}
	if memo[ind][ti][st][f][pd] != 0 {
		return dp[ind][ti][st][f][pd]
	}
	res := int64(0)
	memo[ind][ti][st][f][pd] = 1
	if ti == 1 {
		for i := int64(0); i < num[ind]; i++ {
			if i == 0 {
				res += solve(ind+1, 0, st, f, i)
			}
			if i > 0 {
				if st == 0 {
					res += solve(ind+1, 0, 1, i, i)
				} else {
					res += solve(ind+1, 0, st, f, i)
				}
			}
		}
		if num[ind] == 0 {
			res += solve(ind+1, 1, st, f, 0)
		} else {
			if st == 0 {
				res += solve(ind+1, 1, 1, num[ind], num[ind])
			} else {
				res += solve(ind+1, 1, st, f, num[ind])
			}
		}
	} else {
		for i := int64(0); i < 10; i++ {
			if i == 0 {
				res += solve(ind+1, 0, st, f, i)
			}
			if i > 0 {
				if st == 0 {
					res += solve(ind+1, 0, 1, i, i)
				} else {
					res += solve(ind+1, 0, st, f, i)
				}
			}
		}
	}
	dp[ind][ti][st][f][pd] = res
	return res
}

func main() {
	var l, r int64
	fmt.Scan(&l, &r)

	for r > 0 {
		num = append(num, r%10)
		r /= 10
	}
	n = int64(len(num))
	ans := int64(0)
	for i := range dp {
		for j := range dp[i] {
			for k := range dp[i][j] {
				for m := range dp[i][j][k] {
					for n := range dp[i][j][k][m] {
						dp[i][j][k][m][n] = 0
						memo[i][j][k][m][n] = 0
					}
				}
			}
		}
	}
	ans += solve(0, 1, 0, 0, 0)

	num = nil
	for i := range dp {
		for j := range dp[i] {
			for k := range dp[i][j] {
				for m := range dp[i][j][k] {
					for n := range dp[i][j][k][m] {
						dp[i][j][k][m][n] = 0
						memo[i][j][k][m][n] = 0
					}
				}
			}
		}
	}
	k := l
	for l > 0 {
		num = append(num, l%10)
		l /= 10
	}
	n = int64(len(num))
	ans -= solve(0, 1, 0, 0, 0)

	num = nil
	for k > 0 {
		num = append(num, k%10)
		k /= 10
	}
	if num[0] == num[len(num)-1] {
		ans++
	}
	fmt.Println(ans)
}
